### PR TITLE
petbuilds: add labwc-unstable for experimantal builds

### PR DESCRIPTION
**Pull request by 01micko**
This includes:
- a kitchen sink petbuild including labwc, swaybg, grim, swayidle, swaylock, wlr-randr, slurp, labwc-tweaks, wlopm and snayk (a tiny wayland native game)
- a wallpaper setter based on swaybg that supports multiple outputs
- a wayland color picker supporting hex and rgb() formats
- reasonable default configuration files

### CODE REVIEW

File: woof-code/rootfs-petbuilds/labwc-unstable/petbuild [hunk 1]
```diff
@@ -0,0 +1,103 @@
+download() {
+    [ -f labwc-cfdf60b1da0e23c1dd48aa0a0ad70214c8229fe6.zip ] || wget -t 3 -T 60 -O labwc-cfdf60b1da0e23c1dd48aa0a0ad70214c8229fe6.zip https://github.com/labwc/labwc/archive/cfdf60b1da0e23c1dd48aa0a0ad70214c8229fe6.zip
+    [ -f swaybg-1.1.tar.gz ] || wget -t 3 -T 60 https://github.com/swaywm/swaybg/releases/download/v1.1/swaybg-1.1.tar.gz
+    [ -f grim-1.3.2.tar.gz ] || wget -t 3 -T 60 https://github.com/emersion/grim/releases/download/v1.3.2/grim-1.3.2.tar.gz
+    [ -f swayidle-1.7.tar.gz ] || wget -t 3 -T 60 https://github.com/swaywm/swayidle/releases/download/1.7/swayidle-1.7.tar.gz
+    [ -f swaylock-1.6.tar.gz ] || wget -t 3 -T 60 https://github.com/swaywm/swaylock/releases/download/1.6/swaylock-1.6.tar.gz
```
Comment (dimkr) [inline woof-code/rootfs-petbuilds/labwc-unstable/petbuild:6]: Better use the swaylock petbuild, it has the root patch needed for Puppy

File: woof-code/rootfs-petbuilds/labwc-unstable/petbuild [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    [ -f wlr-randr-v0.2.0.tar.gz ] || wget -t 3 -T 60 -O wlr-randr-v0.2.0.tar.gz https://git.sr.ht/~emersion/wlr-randr/archive/v0.2.0.tar.gz
+    [ -f slurp-bd4b59ef7290345370f7eb591e08103920357af1.zip ] || wget -t 3 -T 60 -O slurp-bd4b59ef7290345370f7eb591e08103920357af1.zip https://github.com/emersion/slurp/archive/bd4b59ef7290345370f7eb591e08103920357af1.zip
+    [ -f labwc-tweaks-34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip ] || wget -t 3 -T 60 -O labwc-tweaks-34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip https://github.com/labwc/labwc-tweaks/archive/34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip
+    if [ ! -f wlopm-27b7ccd1.tar.gz ]; then
```
Comment (dimkr) [inline woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: Better use the wlopm petbuild

Comment (01micko) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: Regarding this and the previous comment, there is a problem with Slackware in that it doesn't ship `wlroots` at all.. users have to roll there own by a slackbuild or whatever. I have an old wlroots-15.1 - old at least for `-current` - pet in the repo, which really is not a great idea. Ideally, the same version of wlroots should accompany labwc and while all this is at he 'unstable' stage  I should really build it here however that really makes it slackware specific, in which case I should rename the petbuild to `labwc-slack`

Here's my untested idea:

```
LWCVER=cfdf60b1da0e23c1dd48aa0a0ad70214c8229fe6
read a b WLRVER <<<$(curl -s https://raw.githubusercontent.com/labwc/labwc/$LWCVER/subprojects/wlroots.wrap|grep 'revision')
WLROOTSVER=${WLRVER:0:7}

download() {
    [ -f seatd-0.5.0.tar.gz ] || wget -t 3 -T 60 -O seatd-0.5.0.tar.gz https://git.sr.ht/~kennylevinsen/seatd/archive/0.5.0.tar.gz
    if [ ! -f wlroots-${WLROOTSVER}.tar.gz ]; then
        git clone --depth 1 https://gitlab.freedesktop.org/wlroots/wlroots.git wlroots-${WLROOTSVER}
        cd wlroots-${WLROOTSVER}
        git checkout ${WLROOTSVER}
        rm -rf .git
        cd ..
        tar -c wlroots-${WLROOTSVER}| gzip -9 > wlroots-${WLROOTSVER}.tar.gz
    fi
    [ -f labwc-$LWCVER.zip ] || wget -t 3 -T 60 -O labwc-$LWCVER.zip https://github.com/labwc/labwc/archive/$LWCVER.zip
    [ -f swaybg-1.1.tar.gz ] || wget -t 3 -T 60 https://github.com/swaywm/swaybg/releases/download/v1.1/swaybg-1.1.tar.gz
    [ -f grim-1.3.2.tar.gz ] || wget -t 3 -T 60 https://github.com/emersion/grim/releases/download/v1.3.2/grim-1.3.2.tar.gz
    [ -f swayidle-1.7.tar.gz ] || wget -t 3 -T 60 https://github.com/swaywm/swayidle/releases/download/1.7/swayidle-1.7.tar.gz
    [ -f swaylock-1.6.tar.gz ] || wget -t 3 -T 60 https://github.com/swaywm/swaylock/releases/download/1.6/swaylock-1.6.tar.gz
    [ -f wlr-randr-v0.2.0.tar.gz ] || wget -t 3 -T 60 -O wlr-randr-v0.2.0.tar.gz https://git.sr.ht/~emersion/wlr-randr/archive/v0.2.0.tar.gz
    [ -f slurp-bd4b59ef7290345370f7eb591e08103920357af1.zip ] || wget -t 3 -T 60 -O slurp-bd4b59ef7290345370f7eb591e08103920357af1.zip https://github.com/emersion/slurp/archive/bd4b59ef7290345370f7eb591e08103920357af1.zip
    [ -f labwc-tweaks-34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip ] || wget -t 3 -T 60 -O labwc-tweaks-34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip https://github.com/labwc/labwc-tweaks/archive/34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip
    if [ ! -f wlopm-27b7ccd1.tar.gz ]; then
        git clone --depth 1 https://git.sr.ht/~leon_plickat/wlopm wlopm-27b7ccd1
        cd wlopm-27b7ccd1
        git checkout 27b7ccd1
        rm -rf .git
        cd ..
        tar -c wlopm-27b7ccd1 | gzip -9 > wlopm-27b7ccd1.tar.gz
    fi
    if [ ! -f snayk-0064ba9a.tar.gz ]; then
        git clone --depth 1 https://git.sr.ht/~leon_plickat/snayk snayk-0064ba9a
        cd snayk-0064ba9a
        git checkout 0064ba9a
        rm -rf .git
        cd ..
        tar -c snayk-0064ba9a | gzip -9 > snayk-0064ba9a.tar.gz
    fi
    [ -f wf-recorder-0.3.0.tar.xz ] || wget -t 3 -T 60 -O wf-recorder-0.3.0.tar.xz https://github.com/ammen99/wf-recorder/releases/download/v0.3.0/wf-recorder-0.3.0.tar.xz
}

build() {
    export PKG_CONFIG_PATH="/usr/lib/pkgconfig:$PKG_CONFIG_PATH"

    tar -xzf seatd-0.5.0.tar.gz
    cd seatd-0.5.0
    meson --buildtype=release --prefix=/usr -Dlogind=disabled -Dbuiltin=enabled -Dserver=disabled -Dexamples=disabled build
    ninja -C build install
    cd ..

    tar -xzf wlroots-${WLROOTSVER}.tar.gz
    cd wlroots-${WLROOTSVER}
    meson --buildtype=release --prefix=/usr -Dxwayland=enabled -Dbackends=drm,libinput build
    ninja -C build install
    cd ..

    unzip labwc-$LWCVER.zip
    cd labwc-$LWCVER
    patch -p1 < ../root.patch
    patch -p1 < ../menu.patch
    meson --buildtype=release --prefix=/usr -Dxwayland=enabled build
    ninja -C build install
    cd ..

    tar -xzf swaybg-1.1.tar.gz
    cd swaybg-1.1
    meson --buildtype=release --prefix=/usr build
    ninja -C build install
    cd ..

    tar -xzf grim-1.3.2.tar.gz
    cd grim-1.3.2
    meson --buildtype=release --prefix=/usr build
    ninja -C build install
    cd ..

    tar -xzf swayidle-1.7.tar.gz
    cd swayidle-1.7
    meson --buildtype=release --prefix=/usr -Dlogind=disabled -Dzsh-completions=false -Dbash-completions=false -Dfish-completions=false build
    ninja -C build install
    cd ..

    tar -xzf swaylock-1.6.tar.gz
    cd swaylock-1.6
    patch -p1 < ../swaylock-root.patch
    meson --buildtype=release --prefix=/usr -Dpam=disabled -Dgdk-pixbuf=enabled -Dzsh-completions=false -Dbash-completions=false -Dfish-completions=false build
    ninja -C build install
    cd ..

    unzip slurp-bd4b59ef7290345370f7eb591e08103920357af1.zip
    cd slurp-bd4b59ef7290345370f7eb591e08103920357af1
    meson --buildtype=release --prefix=/usr build
    ninja -C build install
    cd ..

    unzip labwc-tweaks-34c1f435617c41f9c9cd944d57e0b2585073f8b7.zip
    cd labwc-tweaks-34c1f435617c41f9c9cd944d57e0b2585073f8b7
    meson --buildtype=release --prefix=/usr build
    ninja -C build install
    cd ..

    tar -xzf wlr-randr-v0.2.0.tar.gz
    cd wlr-randr-v0.2.0
    meson --buildtype=release --prefix=/usr build
    ninja -C build install
    cd ..

    tar -xzf wlopm-27b7ccd1.tar.gz
    cd wlopm-27b7ccd1
    make PREFIX=/usr install
    cd ..
    
    tar -xzf snayk-0064ba9a.tar.gz
    cd snayk-0064ba9a
    make PREFIX=/usr install
    cd ..
    
    tar -xJf wf-recorder-0.3.0.tar.xz
    cd wf-recorder-0.3.0
    meson --buildtype=release --prefix=/usr build
    ninja -C build install
    cd ..

    $CC $CFLAGS altf1d.c $LDFLAGS -o /usr/bin/altf1d
    $CC $CFLAGS numlockt.c $LDFLAGS -o /usr/bin/numlockt

    rm -rf /usr/share/wayland-protocols /usr/share/pkgconfig /usr/lib*/pkgconfig /usr/share/aclocal /usr/share/wayland /usr/bin/wayland-scanner
}
```

Comment (dimkr) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: swaylock and wlopem don't need wlroots AFAIK

Comment (01micko) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: I'll do some tests and confirm that, but I still think that I should build wlroots based on the version labwc uses. Makes sense.

Comment (01micko) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: Well, that petbuild I posted works unedited! I just pasted it in verbatim.

Comment (dimkr) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: > but I still think that I should build wlroots based on the version labwc uses

Yeah, sounds like a good idea if it tracks wlroots master.

Comment (dimkr) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: I'd still put labwc+wlroots in one petbuild (and build wlroots with `--default-library=static`), and all other Wayland-related tools (minus those that have their petbuild - wlopm and swaylock) in another.

Comment (01micko) [reply woof-code/rootfs-petbuilds/labwc-unstable/petbuild:10]: As well as swaylock and wlopm, `snayk` can be it's own petbuild too. EDIT: also `swayidle`

**Comment by 01micko** [comment]
Thanks @dimkr for some good ideas. I'll open an issue in the coming days and check stuff off which will make it easier to track stuff.

**Comment by dimkr** [comment]
Feel free to abuse sid, I cleaned up the package list to make it easier to replace yambar with sfwbar and update labwc.

**Pull request closed without merging**